Change way of sensebox/home decoder

mpfeil opened the pull request:
`sensebox/home` profile is now able to handle every sensor setup of a **senseBox:home**.
It`s not restricted to the old full sensor setup. If the `payload is longer than the `sensorMatching` of the correspondig box the unused bytes are ignored. 

## ToDo
- [ ] add test where the `payload` is shorter than the `sensorMatching`

mpfeil committed:
extract possible sensors and build transformer dynamically

mpfeil committed:
add vscode to gitignore

mpfeil committed:
add travis

mpfeil committed:
add Dockerfile and docker-compose for testing

mpfeil committed:
change way of testing

mpfeil committed:
add npm script to run test bash script

mpfeil committed:
sensebox/home profile can be used for every sensor setup

noerw commented on lib/decoding/index.js:
```diff
@@ -50,19 +51,22 @@ const bufferToMeasurements = function bufferToMeasurements (buffer, bufferTransf
-      sensor_id: mask.sensorId,
-      value: mask.transformer(maskedBytes)
-    });
+    if (maskedBytes.length !== 0) {
```
Shouldn't this check `maskedBytes.length === mask.bytes` ?

mpfeil replied:
Good catch. Resolved with last commit

noerw commented on lib/decoding/index.js:
```diff
@@ -72,7 +76,7 @@ const bufferToMeasurements = function bufferToMeasurements (buffer, bufferTransf
   }
 
-  return result;
+  return [result, warnings];
```
why not `{ result, warnings }`? It's more explicit and same amount of chars ;)

mpfeil replied:
Also changed with the last commit

noerw commented on lib/decoding/index.js:
```diff
@@ -169,9 +176,18 @@ const decodeRequest = function decodeRequest (req) {
+
+    // return req.body.payload_fields
+    //   ? decodeJSON(req.body.payload_fields)
+    //   : Promise.reject('no payload for profile `json` provided');
```
These comments can be deleted

noerw reviewed (commented):
Few questions:
- This assumes that measurements are submitted in a specific order, correct? Which order? Would be good to document that.
- Shouldn't the case of to few bytes still be treated as an error? When measurements are missing this is not the behaviour as configured.. Also we don't know which ones are missing, and this can mess up the data.
- How are these warnings handled? afaik there is no mechanism to show them to the user..
- Maybe rebase this onto `feature/mqtt`, as that's the latest code.. But can be done later as well I guess

mpfeil commented:
> This assumes that measurements are submitted in a specific order, correct? Which order? Would be good to document that.

Yes of course. If we merge the PR there will also be a generated [LoRa sketch](https://github.com/sensebox/node-sketch-templater/blob/lora/templates/homev2_lora.tpl) at the end of registration. But afaik the specific order is always relevant. Or did I miss something?

> Shouldn't the case of to few bytes still be treated as an error? When measurements are missing this is not the behaviour as configured.. Also we don't know which ones are missing, and this can mess up the data.

Fair point. I will think about it.

> How are these warnings handled? afaik there is no mechanism to show them to the user..

Right now there is no mechanism. But there will be later. Just as a preparation for it.

> Maybe rebase this onto feature/mqtt, as that's the latest code.. But can be done later as well I guess

We will do it later. We want to release the new LoRa sketch generation and after it release your changes of feature/mqtt

mpfeil committed:
return an object and delete unused comments

mpfeil committed:
fix bytes check

mpfeil committed:
use node-config module for configuration

changed to node config module in preparation to use
@sensebox/opensensemap-api-models module

mpfeil committed:
install @sensebox/opensensemap-api-models and fix imports

mpfeil committed:
resolve all errors occured by changing to @sensebox/opensensemap-api-models

mpfeil committed:
use beta release of opensensemap-api-models

mpfeil committed:
populate box with last measurements

mpfeil committed:
use api models version 0.0.11

mpfeil committed:
update configuration section in readme

Commit:
use models v0.0.12

felixerdy committed:
add new sensors to homedecoder

felixerdy committed:
bump opensensemap-api-models version

felixerdy committed:
bump docker node version to 12

felixerdy committed:
use node 12 in test dockerfile

felixerdy committed:
fix spelling mistake

felixerdy committed:
update documentation 📜

felixerdy committed:
also include boxes with null port

felixerdy committed:
add windspeed to home decoder

mpfeil commented:
Closed. Already included in new `main` branch
